rasa/core/policies/intent_ted_policy.py

```diff
@@ -0,0 +1,691 @@
+
+        Args:
+            tracker: Tracker containing past conversation events.
+            domain: Domain of the assistant.
+            interpreter: Interpreter which may be used by the policies to create
+                additional features.
+
+        Returns:
+             The policy's prediction (e.g. the probabilities for the actions).
+        """
+        if self.model is None:
+            return self._prediction(self._default_predictions(domain))
+
+        # Prediction through the policy is skipped if:
+        # 1. Last event in the tracker was not of type `UserUttered`.
+        # This is to prevent the ensemble of policies from being stuck
+        # in a loop.
+        # 2. If the tracker does not contain any event of type `UserUttered` till now.
+        if not tracker.get_last_event_for(UserUttered) or (
+            tracker.events and not isinstance(tracker.events[-1], UserUttered)
+        ):
+            logger.debug(
+                f"Skipping predictions for {self.__class__.__name__} "
+                f"as the last event in tracker is not of type `UserUttered`."
+            )
+            return self._prediction(self._default_predictions(domain))
+
+        # create model data from tracker
+        tracker_state_features = self._featurize_for_prediction(
+            tracker, domain, interpreter
+        )
+
+        model_data = self._create_model_data(tracker_state_features)
+        output = self.model.run_inference(model_data)
+
+        # take the last prediction in the sequence
+        similarities = output["similarities"][:, -1, :]
+
+        # Check for unlikely intent
+        query_intent = tracker.get_last_event_for(UserUttered).intent_name
+        is_unlikely_intent = self._check_unlikely_intent(
+            domain, similarities, query_intent
+        )
+
+        confidences = list(np.zeros(domain.num_actions))
+
+        if is_unlikely_intent:
+            confidences[domain.index_for_action(ACTION_UNLIKELY_INTENT_NAME)] = 1.0
+
+        return self._prediction(
+            confidences,
+            action_metadata=self._collect_action_metadata(domain, similarities),
+        )
+
+    def _should_check_for_intent(self, intent: Text, domain: Domain) -> bool:
+        """Check if the intent should raise `action_unlikely_intent`.
+
+        Args:
+            intent: Intent to be queried.
+            domain: Domain of the assistant.
+
+        Returns:
+            Whether intent should raise `action_unlikely_intent` or not.
+        """
+        if domain.intents.index(intent) not in self.label_thresholds:
+            # This means the intent was never present in a story
+            logger.debug(
+                f"Query intent index {domain.intents.index(intent)} not "
+                f"found in label thresholds - {self.label_thresholds}"
+            )
+            return False
+        if intent in self.config[IGNORE_INTENTS_LIST]:
+            return False
```

Comment:
Shouldn't this check occur first, to avoid showing the above debug message for an ignored intent?

Reply:
Hmm, I think the earlier debug message is necessary just to communicate which intent is being queried. However, I think it would be better if I add another debug message inside the above `if` construct, something along the lines of - 
`Query intent {intent} found in {IGNORE_INTENTS_LIST}. Check for `action_unlikely_intent` prediction will be skipped.` What do you think?

Reply:
Wouldn't you never get this second message? Because it would return `False` beforehand?

Reply:
I meant adding the above debug message before `return False` but inside `if intent in self.config[IGNORE_INTENTS_LIST]:`

Reply:
Yes, I meant even higher up than that. If the intent is in `IGNORE_INTENTS_LIST` wouldn't this:
`if domain.intents.index(intent) not in self.label_thresholds:`
be true?

Reply:
Nope. `label_thresholds` are learnt during training for all intents that are seen across training stories.
`IGNORE_INTENTS_LIST` is primarily meant to be used at inference time where users can set it to a subset of intents for which they don't want to trigger `action_unlikely_intent`.

A follow up question from your side could be - "Can we even change the component's configuration parameters without retraining the pipeline?" Answer: Not currently, but something to be prepared for in 3.0 . Once that's possible, users can change `IGNORE_INTENTS_LIST` and play with the policy to get the best set of intents for which they want to trigger `action_unlikely_intent` without having to retrain the pipeline.
Does that help clarify?

Reply:
Yes, thank you!